Formula/jdtls.rb

```diff
@@ -0,0 +1,32 @@
+class Jdtls < Formula
+  desc "Java language specific implementation of the Language Server Protocol"
+  homepage "https://github.com/eclipse/eclipse.jdt.ls"
+  url "https://download.eclipse.org/jdtls/milestones/1.7.0/jdt-language-server-1.7.0-202112161541.tar.gz"
+  sha256 "2f0c28dfec317a268ec44904420657181b43a7ba2a32f0bf788ea388dacb8552"
+  license "EPL-2.0"
+
+  depends_on "openjdk"
+
+  def install
+    libexec.install "config_mac"
+    libexec.install "features"
+    libexec.install "plugins"
+
+    (buildpath/"jdtls").write <<~BASH
+      #!/bin/bash
+      # copy the configuration folder to tmp to be writable
+      tmp_dir="$(mktemp -d /tmp/jdtls.XXXXX)"
+      cp -R #{libexec}/config_mac "${tmp_dir}"
+      # and ensure that it is removed on exit
+      trap "{ rm -rf ${tmp_dir}; }" EXIT
+
+      java -Declipse.application=org.eclipse.jdt.ls.core.id1 \\
+        -Dosgi.bundles.defaultStartLevel=4 \\
+        -Declipse.product=org.eclipse.jdt.ls.core.product \\
+        -Xms1G -jar #{libexec}/plugins/org.eclipse.equinox.launcher_*.jar \\
+        -configuration "${tmp_dir}/config_mac" "$@"
+    BASH
+
+    bin.install "#{buildpath}/jdtls"
+  end
+end
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
I'll look into this

Reply:
Test is working now but the CI complains about bottles. With a bottle (copied from Kotlin language server) I get: `New formulae in homebrew/core should not have a `bottle do` block` and without a bottle I get an error that it cannot be build without a bottle. I'm confused.

Reply:
That hit me again.